benches/micro/src/benches/mod.rs

```diff
@@ -117,3 +126,101 @@ pub fn bench_sched_two_threads() -> Result<(), ()> {
 
 	Ok(())
 }
+
+// derived from
+// https://github.com/rust-lang/compiler-builtins/blob/master/testcrate/benches/mem.rs
+fn memcpy_builtin(n: usize) {
+	let v1 = vec![1u8; n];
+	let mut v2 = vec![0u8; n];
+
+	let now = Instant::now();
+	for _i in 0..NR_RUNS {
+		let src: &[u8] = black_box(&v1);
+		let dst: &mut [u8] = black_box(&mut v2);
+		dst.copy_from_slice(src);
+	}
+
+	println!(
+		"memcpy_builtin:  {} block size, {} MByte/s",
+		n,
+		(NR_RUNS * n) as f64 / (1024.0 * 1024.0 * now.elapsed().as_secs_f64())
+	);
+}
+
+// derived from
+// https://github.com/rust-lang/compiler-builtins/blob/master/testcrate/benches/mem.rs
+fn memset_builtin(n: usize) {
+	let mut v1 = vec![0u8; n];
+	let now = Instant::now();
+	for _i in 0..NR_RUNS {
+		let dst: &mut [u8] = black_box(&mut v1);
+		let val: u8 = black_box(27);
+		for b in dst {
+			*b = val;
+		}
+	}
+
+	println!(
+		"memset_builtin:  {} block, {} MByte/s",
+		n,
+		((NR_RUNS * n) >> 20) as f64 / now.elapsed().as_secs_f64()
+	);
+}
+
+// derived from
+// https://github.com/rust-lang/compiler-builtins/blob/master/testcrate/benches/mem.rs
+fn memcpy_rust(n: usize) {
+	let v1 = vec![1u8; n];
+	let mut v2 = vec![0u8; n];
+	let now = Instant::now();
+	for _i in 0..NR_RUNS {
+		let src: &[u8] = black_box(&v1[0..]);
+		let dst: &mut [u8] = black_box(&mut v2[0..]);
+		unsafe {
+			memcpy(
+				dst.as_mut_ptr() as *mut c_void,
+				src.as_ptr() as *mut c_void,
+				n,
+			);
+		}
+	}
+
+	println!(
+		"memcpy_rust:  {} block, {} MByte/s",
+		n,
+		((NR_RUNS * n) >> 20) as f64 / now.elapsed().as_secs_f64()
+	);
+}
+
+// derived from
+// https://github.com/rust-lang/compiler-builtins/blob/master/testcrate/benches/mem.rs
+fn memset_rust(n: usize) {
+	let mut v1 = vec![0u8; n];
+	let now = Instant::now();
+	for _i in 0..NR_RUNS {
+		let dst: &mut [u8] = black_box(&mut v1[0..]);
+		let val = black_box(27);
+		unsafe {
+			memset(dst.as_mut_ptr() as *mut c_void, val, n);
+		}
+	}
+
+	println!(
+		"memset_rust:  {} block, {} MByte/s",
+		n,
+		((NR_RUNS * n) >> 20) as f64 / now.elapsed().as_secs_f64()
+	);
+}
+
+pub fn bench_mem() -> Result<(), ()> {
+	memcpy_builtin(4096);
+	memcpy_builtin(1048576);
+	memset_builtin(4096);
+	memset_builtin(1048576);
+	memcpy_rust(4096);
+	memcpy_rust(1048576);
+	memset_rust(4096);
+	memset_rust(1048576);
```

Comment:
4KB and 1MB all fit into the cache of modern CPUs. Ideally we would also benchmark with a size that is twice the size of the last level cache of the system.

Reply:
Ok, I also also add a benchmark with a larger array size.